### PR TITLE
statistics

### PR DESCRIPTION
**Pull request by n1tr0-5urf3r**
Added $statistics/stats user/guild/total

### CODE REVIEW

File: cogs/statistics.py [hunk 1]
```diff
@@ -12,6 +14,137 @@ def __init__(self, bot):
...
+        try:
+            await ctx.send(embed=data)
+        except discord.Forbidden:
+            await ctx.send("Need permission to embed links")
+
+    @statistics.command(name="guild")
+    async def statistics_guild(self, ctx):
+        """Statistics of this guild
+
+        Only available on Discord Server"""
```
Comment (Maselkov) [inline cogs/statistics.py]: Add a 
```py
@commands.guild_only()
```
decorator and you can get rid of this bit, and the extended help command.

File: cogs/statistics.py [hunk 2]
```diff
@@ -12,6 +14,137 @@ def __init__(self, bot):
...
+            await ctx.send(embed=data)
+        except discord.Forbidden:
+            await ctx.send("Need permission to embed links")
+
+    @statistics.command(name="total")
+    @commands.is_owner()
+    async def statistics_total(self, ctx):
+        """Total stats of the bot's commands
+
+        Only available to server owner"""
```
Comment (Maselkov) [inline cogs/statistics.py:57]: is_owner refers to bot owner

File: cogs/statistics.py [hunk 3]
```diff
@@ -12,6 +14,137 @@ def __init__(self, bot):
...
+        # Get data
+        total_amount = await cursor.count()
+        ordered_commands = await self.get_commands(cursor, 'command')
+        percentages = self.calc_percentage(ordered_commands, total_amount)
+        data.add_field(
+            name="Total commands", value=str(total_amount), inline=False)
+        output = self.generate_commands(ordered_commands)
+        data.add_field(name="Most used commands", value=output, inline=False)
+        output = self.generate_diagramm(percentages)
+        data.add_field(name="Diagramm", value=output, inline=False)
```
Comment (Maselkov) [inline cogs/statistics.py]: Typo: `diagram`

File: cogs/statistics.py [hunk 4]
```diff
@@ -12,6 +14,137 @@ def __init__(self, bot):
...
+    def calc_percentage(self, ordered_commands, total):
+        """Generates ordered dict of percentages of used commands from ordered_commands"""
+        percentages = {}
+        for k, v in ordered_commands.items():
+            percentages[k] = round(100 / total * v)
+        ordered_percentages = collections.OrderedDict(
+            sorted(percentages.items(), key=lambda x: x[1], reverse=True))
+        return ordered_percentages
+
+    async def generate_embed(self, ctx, data, cursor, rank=True):
```
Comment (Maselkov) [inline cogs/statistics.py]: Method name is a big vague, rank should probably be a keyword only argument

File: cogs/statistics.py (hunk 1 above)
Comment (n1tr0-5urf3r) [reply cogs/statistics.py]: Added decorator, removed check for guild_id

File: cogs/statistics.py (hunk 2 above)
Comment (n1tr0-5urf3r) [reply cogs/statistics.py:57]: Updated helper message

File: cogs/statistics.py (hunk 3 above)
Comment (n1tr0-5urf3r) [reply cogs/statistics.py]: Changed typo, also refactored generate_diagram()

File: cogs/statistics.py (hunk 4 above)
Comment (n1tr0-5urf3r) [reply cogs/statistics.py]: changed to     async def generate_embed(self, ctx, data, cursor, *, rank=True):

**Review by n1tr0-5urf3r — COMMENTED** [review]
Requested changes

File: cogs/statistics.py [hunk 5]
```diff
@@ -12,6 +14,136 @@ def __init__(self, bot):
...
+        """Generates string of ASCII bar out of ordered_dict of percentages"""
+        counter = 0
+        output = ""
+        for k, v in percentages.items():
+            if counter < 5:
+                bar_count = round(v / 5)
+                for emoji in range(0, bar_count):
+                    output += "▓"
+                tab_count = 20 - bar_count
+                for tab in range(0, tab_count):
```
Comment (Maselkov) [inline cogs/statistics.py]: Style: range() doesn't need the initial 0, it defaults to it
```py
range(tab_count)
```

Comment (n1tr0-5urf3r) [reply cogs/statistics.py]: Removed 0 from range()

**Pull request merged**